### ISSUE TITLE
Consider adding a const version of `blockdata::genesis_block_header`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2645]
Requires const'ing a few other things and hardcoding the merkle root but looks like its doable without too much trouble.

**Comment by apoelstra** [comment]
Yeah, it's likely not done because of the merkle root computation. Not sure if we have the necessary const constructors for hashes, even now.

**Comment by Kixunil** [comment]
Something definitely needs to be hardcoded because you can't create `Transaction` in const context as it has non-empty `Vec`s.

We can hardcode coinbase `Txid` which IIRC equals merkle root.

**Comment by apoelstra** [comment]
Let's change this issue to "add a const `blockdata::genesis_block_header` then.

I guess one day in the far-far future maybe you will be able to create `Vec`s in a const-context. It is an opaque type so in principle they could have a completely different representation that they swap in for constfns, or something.

**Comment by Kixunil** [comment]
I don't think it'll be possible to assign it to consts though. Anyway yes, `block_header` then.
